src/utils/daos_hdlr.c

```diff
@@ -1535,6 +1536,741 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
 	return rc;
 }
 
+static int
+parse_filename(const char* path, char** _obj_name, char** _cont_name)
+{
+	char *f1 = NULL;
+	char *f2 = NULL;
+	char *fname = NULL;
+	char *cont_name = NULL;
+	int rc = 0;
+	if (path == NULL || _obj_name == NULL || _cont_name == NULL)
+			return -EINVAL;
+
+	if (strcmp(path, "/") == 0) {
+		*_cont_name = strdup("/");
+		if (*_cont_name == NULL)
+			return -ENOMEM;
+		*_obj_name = NULL;
+		return 0;
+	}
+	f1 = strdup(path);
+	if (f1 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+
+	f2 = strdup(path);
+	if (f2 == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+	fname = basename(f1);
+	cont_name = dirname(f2);
+
+	if (cont_name[0] == '.' || cont_name[0] != '/') {
+		char cwd[1024];
+
+		if (getcwd(cwd, 1024) == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+
+		if (strcmp(cont_name, ".") == 0) {
+			cont_name = strdup(cwd);
+			if (cont_name == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+		} else {
+			char *new_dir = calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+			if (new_dir == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+
+			strcpy(new_dir, cwd);
+			if (cont_name[0] == '.') {
+				strcat(new_dir, &cont_name[1]);
+			} else {
+				strcat(new_dir, "/");
+				strcat(new_dir, cont_name);
```

Comment:
I wonder if we can utilize `snprintf` here for the `strcpy` and `strcat` for performance. (I have not tested)

(Again, this similar function exists elsewhere)

Reply:
@daltonbohning yes, this parse_filename function is also in mpifileutils. I think that one still uses the strcpy and strcat as well though.

Reply:
Yeah, it does. I think we can probably leave this function as-is for now, until we figure out how we're going to centralize the code. I believe this function in mpifileutils was also copied from somewhere else, so we have at least 3 copies/verisons that I''m aware of.

Reply:
more than that :-)
i've written this function originally for the mpiio driver in mpich. now it's also in ior and HDF5.